jgrapht-core/src/main/java/org/jgrapht/GraphMetrics.java

```diff
@@ -207,6 +209,114 @@
                 && (graph.getType().isDirected() || graph.getType().isAllowingMultipleEdges());
         return girth;
     }
+
+    /**
+     * An $O(|V|^3)$ naive implementation for counting non-trivial triangles in an undirected graph
+     * induced by the subset of vertices.
+     *
+     * @param graph the input graph
+     * @param vertexSubset the vertex subset
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return the number of triangles in the graph induced by vertexSubset
+     */
+    private static <V, E> long naiveCountTriangles(Graph<V, E> graph, List<V> vertexSubset){
```

Comment:
Since you mention that it is an $n^3$ algorithm, maybe you should also state in the documentation that the `vertexSubset` should provide constant time indexing.